src/scenario/scenario_events_xml.h

```diff
@@ -0,0 +1,638 @@
+                                        .xml_parm1 =    { .name = "resource",       .type = PARAMETER_TYPE_RESOURCE, },
+                                        .xml_parm2 =    { .name = "amount",         .type = PARAMETER_TYPE_NUMBER,           .min_limit = 0,           .max_limit = 100000 },
+                                        .xml_parm3 =    { .name = "set_buy_price",  .type = PARAMETER_TYPE_BOOLEAN,          .min_limit = 0,           .max_limit = 1 },
+                                        .xml_parm4 =    { .name = "show_message",   .type = PARAMETER_TYPE_BOOLEAN,          .min_limit = 0,           .max_limit = 1 },
+                                        .xml_parm5 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED }, },
+    [ACTION_TYPE_EMPIRE_MAP_CONVERT_FUTURE_TRADE_CITY]     = { .type = ACTION_TYPE_EMPIRE_MAP_CONVERT_FUTURE_TRADE_CITY,
+                                        .xml_attr =     { .name = "empire_map_convert_future_trade_city",     .type = PARAMETER_TYPE_TEXT },
+                                        .xml_parm1 =    { .name = "target_city",    .type = PARAMETER_TYPE_FUTURE_CITY },
+                                        .xml_parm2 =    { .name = "show_message",   .type = PARAMETER_TYPE_BOOLEAN,          .min_limit = 0,           .max_limit = 1 },
+                                        .xml_parm3 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
+                                        .xml_parm4 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
+                                        .xml_parm5 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED }, },
+    [ACTION_TYPE_REQUEST_IMMEDIATELY_START]     = { .type = ACTION_TYPE_REQUEST_IMMEDIATELY_START,
+                                        .xml_attr =     { .name = "request_immediately_start",     .type = PARAMETER_TYPE_TEXT },
+                                        .xml_parm1 =    { .name = "request_id",     .type = PARAMETER_TYPE_NUMBER,           .min_limit = 0,           .max_limit = 19 },
+                                        .xml_parm2 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
+                                        .xml_parm3 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
+                                        .xml_parm4 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
+                                        .xml_parm5 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED }, },
+};
+
+typedef struct {
+    parameter_type type;
+    const char *text;
+    int value;
+} special_attribute_mapping_t;
+
+#define SPECIAL_ATTRIBUTE_MAPPINGS_CHECK_SIZE 3
+static special_attribute_mapping_t special_attribute_mappings_check[SPECIAL_ATTRIBUTE_MAPPINGS_CHECK_SIZE] = {
+    [0]     = { .type = PARAMETER_TYPE_CHECK,                .text = "eq",                            .value = COMPARISON_TYPE_EQUAL},
```

Comment:
```suggestion
    { .type = PARAMETER_TYPE_CHECK,                .text = "eq",                            .value = COMPARISON_TYPE_EQUAL},
```

Since these are normal array values and not enums, I don't think you need to use the `[0] =` here. Same applies to all below cases.

Reply:
Done.